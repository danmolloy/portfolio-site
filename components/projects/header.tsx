import { FaLaptopCode, FaLink } from "react-icons/fa"

export type ProjectHeaderProps = {
  title: string
  githubLink: string
  previewLink: string
  selectedTab: number
  setSelectedTab: (arg: number) => void
}

export default function ProjectHeader(props: ProjectHeaderProps) {
  const { githubLink, previewLink, selectedTab, setSelectedTab, title } = props;

  return (
    <div data-testid="project-header" className=" w-full">
      <div className="flex flex-row justify-between items-start w-full">
      <h2 className="text-blue-700 dark:text-blue-400">{title}</h2>
      <div className="flex flex-col">
        <a data-testid="github-link" href={githubLink} target="_blank" className="m-1 ml-2  flex flex-row items-center text-sm hover:text-blue-500 dark:hover:text-yellow-300">
          <FaLaptopCode />
          <p className="ml-1 hover:text-black dark:hover:text-yellow-300">
            View GitHub
          </p>
        </a>
        <a data-testid="example-link" href={previewLink} target="_blank" className="m-1 flex flex-row items-center text-sm hover:text-blue-500 dark:hover:text-yellow-300">
          <FaLink />
          <p className="ml-1 hover:text-black dark:hover:text-yellow-300">
            Live Example
          </p>
        </a>
        </div>
        </div>
      <div className="w-full flex flex-row mb-6">
        <button data-testid="overview-btn" className={selectedTab === 0 ? "custom-border-gradient border-b border-blue-500  py-2 my-2 mx-4 hover:text-zinc-400 " : "py-2 my-2 mx-4 hover:text-zinc-400"} onClick={() => setSelectedTab(0)}>Overview</button>
        <button className={selectedTab === 1 ? "custom-border-gradient border-b border-blue-500  py-2 my-2 mx-4 hover:text-zinc-400 " : "py-2 my-2 mx-4 hover:text-zinc-400" } onClick={() => setSelectedTab(1)}>Features</button>
      </div>
    </div>
  )
}