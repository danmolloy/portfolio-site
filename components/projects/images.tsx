import Image from "next/image"
import { useState } from "react"
import { FaChevronRight, FaChevronLeft } from "react-icons/fa";
import { BsDot } from "react-icons/bs";

export type ProjectImagesProps = {
  images: {
    imgSrc: string
    imgWidth: string
    imgHeight: string
    imgAlt: string
  }[]
}

export default function ProjectImages(props: ProjectImagesProps) {
  const { images } = props
  const [currentElIndex, setCurrentElIndex] = useState<number>(0)
  const [awaitChange, setAwaitChange] = useState<boolean>(false)

  const handleChevronClick = (newIndex: number) => {
    setAwaitChange(true)
    setTimeout(() => {
      setCurrentElIndex(newIndex)
      setAwaitChange(false)
    }, 300)
  }

  return (
    <div data-testid="project-images" className={` m-4 h-[${Number(images[currentElIndex].imgHeight)}px]  flex flex-col items-center justify-center `}>
      <Image 
          priority={true} key={images[currentElIndex].imgSrc} 
          className={awaitChange ? "mx-16 opacity-0 transition-opacity duration-200" : "mx-16 fade-in transition-opacity duration-200"} 
          alt={images[currentElIndex].imgAlt} src={images[currentElIndex].imgSrc} width={Number(images[currentElIndex].imgWidth)/4} height={Number(images[currentElIndex].imgHeight)/4} />
      <div className="flex flex-row my-4">
      <button
        data-testid="left-chevron"
        className={currentElIndex === 0 ? "opacity-30 transition-opacity duration-500 text-xl hover:text-amber-500": "opacity-100 transition-opacity duration-500 text-xl hover:text-amber-500"}
          onClick={() => {
          currentElIndex > 0 
          && handleChevronClick(currentElIndex - 1)
        }}>
        <FaChevronLeft />
      </button>
      <div className="flex flex-row">
        {images.map((i, index) => (
          <button key={i.imgSrc} onClick={() => handleChevronClick(index)} className={`hover:text-yellow-400 ${currentElIndex === index && "text-yellow-400"}  text-2xl`}>
            <BsDot />
          </button>
        ))}
      </div>
      <button 
        data-testid="right-chevron"
        className={
          currentElIndex >= images.length - 1 
          ? "opacity-30 transition-opacity duration-700 text-xl hover:text-amber-500"
          : "opacity-100 transition-opacity duration-700 text-xl hover:text-amber-500"}
        onClick={() => {
          currentElIndex < images.length - 1 
          && handleChevronClick(currentElIndex + 1)
        }}>
        <FaChevronRight />
      </button>
      </div>
    </div>
  )
}