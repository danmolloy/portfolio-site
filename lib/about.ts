export const aboutData: {
  snippets: string[]
  orchestras: {
    name: string
    favicon: string
  }[]
} = {
  snippets: [ 
    "I have spent my whole professional life working as a classical double bassist, and regularly perform with some of the finest ensembles in the world. I frequently work in the renowned Abbey Road Studios with the top session musicians in UK, recording soundtracks for award winning films and television series.",
    "I have enjoyed writing code for the past four years as a recreational activity and am looking to make it a career.", 
    "Born in Brisbane, I have lived in London for 9 years.",
  ],
  orchestras: [
    {
      name: "Academy of St Martin in the Fields",
      favicon: "/icons/asmf.png"
    },
    {name: "BBC Symphony Orchestra", favicon: "/icons/bbc.png"},
    {name: "City of London Sinfonia", favicon: "/icons/cls.png"},
    {name: "Cool Music Ltd", favicon: "/icons/cool-music.jpg"},
    {name: "English Chamber Orchestra", favicon: "/icons/eco.png"},
    {name: "Isobel Griffiths Ltd", favicon: "/icons/ig-ltd.png"},
    {name: "London Chamber Orchestra", favicon: "/icons/lco.png"},
    {name: "London Mozart Players", favicon: "/icons/lmp.png"},
    {name: "London Sinfonietta", favicon: "/icons/london-sinfonietta.jpg"},
    {name: "London Contemporary Orchestra", favicon: "/icons/london-contemporary-orchestra.png"},
    {
      name: "London Symphony Orchestra",
      favicon: "/icons/lso.png"
    },
    {name: "Orchestra of the Swan", favicon: "/icons/OOTS.png"},
    {name: "Oxford Philharmonic", favicon: "/icons/opo.png"},
    {name: "Queensland Symphony Orchestra", favicon: "/icons/qso.jpg"},
    {name: "Royal Philharmonic Orchestra", favicon: "/icons/rpo.jpg"},
    ]
}