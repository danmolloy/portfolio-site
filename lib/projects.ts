export type ProjectDivProps = {
  id: number
  title: string
  githubLink: string
  previewLink: string
  images: {
    imgSrc: string
    imgWidth: string
    imgHeight: string
    imgAlt: string
  }[]
  overview: string[],
  features: string[]
  stack: string[]
}

export const projectsBlurb = "All projects are built with Test-Driven Development, RESTful API Integration, responsive mobile-first design and W3C Accessibility."


export const projectsArr: ProjectDivProps[] = [
  {
    id: 4,
    title: "TaskTrek",
    previewLink: "https://apps.apple.com/gb/app/tasktrek/id6462425051",
    githubLink: "https://github.com/danmolloy/routinee",
    overview: [
      "I enjoy learning new skills and am keen on maintaining habits which help achieve larger goals. TaskTrek helps the user log their daily habits with additional data (i.e. duration, count, hashtags) and view an overall pattern across a week, month or year. I created TaskTrek for my own use and eventually released it on the App Store.",
      "TaskTrek is built with React Native, TypeScript, Expo, Jest. Additionally, I used a few RN community packages to a lesser extent."
    ],
    images: [
      {imgWidth: "750",
      imgHeight: "1334",
      imgSrc: '/images/tasktrek-home.PNG',
      imgAlt: "App home page of TaskTrek"},
      {imgWidth: "750",
      imgHeight: "1334",
      imgSrc: '/images/tasktrek-detail.PNG',
      imgAlt: "TaskTrek activity detail page featuring github-style heat chart"},
      {imgWidth: "750",
      imgHeight: "1334",
      imgSrc: '/images/tasktrek-help.PNG',
      imgAlt: "TaskTrek help page featuring encouraging words"},
    ],
    features: [
        "CRUD database operations for activities and instances of activity",
        "Year overview heat map of activity, with option to filter by hashtag",
        "Interactive home page with haptics and animation when ticking of activity",
        "Animated movements on screen changes for a game-like experience",
      ],
    stack: [
        "React Native",
        "TypeScript",
        "Jest",
        "Expo"
      ],
  },
  {
    id: 0,
    title: "Musician Website",
    previewLink: 'https://www.fionakellyflute.com/',
    githubLink: 'https://github.com/danmolloy/fio-kelly',
    images: [
      {imgSrc: '/images/fk-landing.png',
      imgWidth: '870',
      imgHeight: '480',
      imgAlt: "Hero section of website"},
      {imgSrc: '/images/fk-albums.png',
      imgWidth: '870',
      imgHeight: '480',
      imgAlt: "Array of album covers"},
      {imgSrc: '/images/fk-contact.png',
      imgWidth: '870',
      imgHeight: '480',
      imgAlt: "Contact section and footer"},
    ],
    overview: [
      "I was approached by a colleague to create a website for prospective students, potential employers and concert organisers. The musician wanted a site that was modern and attractive, but largely gave me freedom to design the interface. I worked closely with them to create a design they were pleased with. It's not the most technical project in my portfolio but very practical.",
      "This site was built with NextJS, TypeScript and TailwindCSS. The contact form was built using NodeMailer, Formik and Yup validation."
  ],
    features: [
        "Event listeners to make header disappear on scroll",
        "Page scrolls to section on menu link click",
        "Attractive, modern design with logical layout",
        "Nodemailer contact form with email integration"
      ],
    stack: [
        "Next.js",
        "TypeScript",
        "TailwindCSS",
      ],
  },
  {
    id: 1,
    title: "Spelling Bee",
    previewLink: 'https://spelling-bee-danmolloy.vercel.app',
    githubLink: 'https://github.com/danmolloy/spelling-bee',
    images: [
      {imgSrc: '/images/spelling-bee.PNG',
      imgWidth: '750',
      imgHeight: '1092',
      imgAlt: "Spelling Bee screenshot with pangram confetti"
},
{imgSrc: '/images/spelling-bee-list.PNG',
imgWidth: '750',
imgHeight: '1092',
imgAlt: "Found words"},
{imgSrc: '/images/spelling-bee-yoyo.PNG',
imgWidth: '750',
imgHeight: '1092',
imgAlt: "Game interface"},
    ],
    overview: [
      "I'm a great fan of NYT's Spelling Bee word game and decided to replicate the game. The site fetches the original game data from the source code and saves your answers to local storage, allowing you to continue gameplay throughout the day. I&apos;ve tried my best to capture the playful character of the game using CSS animation and styling. Additionally, there is a dynamic table of hints.",
      "Spelling Bee is built with NextJS, TypeScript, TailwindCSS (as well as vanilla CSS) and Jest. For the data fetching, I use NextJS API routes."
    ],
      
    features: [
      "Replica of NYT original interface, including hexagonal 'hive'",
      "Playful CSS animations, in keeping with the game character",
      "Keyboard event listeners",
      "Fetches original game data from NYT",
      "Local storage saves user progress",
      "Dynamic hints page gives count for word combinations",
      ],
    stack: [
        "Next.js",
        "TypeScript",
        "TailwindCSS",
        "Jest"
      ],
  },
  {
    id: 2,
    title: "GigFix",
    previewLink: 'https://gig-fix-git-main-danmolloy.vercel.app/',
    githubLink: 'https://github.com/danmolloy/fixer',
    images: [
      {imgSrc: '/images/gig-fix.PNG',
      imgWidth: '750',
      imgHeight: '1092',
      imgAlt: "GigFix landing page"},
      {imgSrc: '/images/gig-fix-calendar.PNG',
      imgWidth: '750',
      imgHeight: '1092',
      imgAlt: "GigFix calendar interface"},
      {imgSrc: '/images/gig-fix-directory.PNG',
      imgWidth: '750',
      imgHeight: '1092',
      imgAlt: "GigFix player directory"},
    ],
    overview: [
      "Orchestras often need to book a large number of freelancers in a very short space of time. Additionally, they must relay dynamic information to all relative musicians. GigFix address this workflow issue, saving orchestras large amounts of time and money. For the freelance musicians, it offers a directory where they can be connected with orchestras with whom they’ve not worked. I’m hoping to eventually release GigFix for public use.",
      "GIgFix is built primarily with NextJS, TypeScript, Formik,  Prisma (with PostgresQL), Twilio, TailwindCSS and Jest.",
    ],
    features: [
      "Attractive external site and landing page",
      "Intuitive calendar interface for an overview of events in month, week or day view",
      "A variety of CRUD database operations for a relational database",
      "Text messaging API integration",
      "Social network user profiles & directory",
      "Notifications section in the style of GitHub notifications page",
      "Concise event page information layout",
      "Flexible booking system for a variety of needs (i.e. only selected rehearsals, availability check vs booking, additional messages to individual)",
      "Simple address book layout for concise information display",
      ],
    stack: [
        "Next.js",
        "TypeScript",
        "Node.js",
        "TailwindCSS",
        "Jest",
        "Prisma",
        "PostgreSQL",
        "Twilio",
        "Formik"
      ],
  },
  {
    id: 3,
    title: "MyJournal",
    previewLink: 'https://django-journal-production.up.railway.app/',
    githubLink: 'https://github.com/danmolloy/django-journal',
    images: [
      {imgSrc: '/images/myjournal-index.png',
      imgWidth: '1000',
      imgHeight: '1100',
      imgAlt: "Index View"},
      {imgSrc: '/images/myjournal-detail.png',
      imgWidth: '1000',
      imgHeight: '1100',
      imgAlt: "Entry Detail View"},
      {imgSrc: '/images/myjournal-signin.png',
      imgWidth: '1000',
      imgHeight: '1100',
      imgAlt: "Edit Entry View"},
    ],
    overview: [
      "My journal is a basic journalling web app, created as a personal learning tool when I was studying Django and Python. Very simply, users create an account and create journal entries which are only visible to themselves.",
      "MyJournal is built with Python and Django. It connects to a SQLite3 database and uses Django packages such as Django Auth."
    ],
    features: [
      "Authentication and authorisation",
      "All CRUD operations on journal entries",
      "Homepage, detail and sign in/sign up views", 
      "Password reset",
      "Vanilla CSS styling of Django views",
      ],
    stack: [
        "Python",
        "Django",
        "Django-Templates",
        "PostgreSQL",
      ],
    
  }
]