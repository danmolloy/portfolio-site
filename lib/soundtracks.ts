
export type Recording = {
  title: string
  year: number
  imgSrc: string
  imgWidth: number
  imgHeight: number
}

export const soundtracks: Recording[] = [
  {title:"The Crown", year: 2021, imgSrc: "/soundtracks/thecrown.jpg", imgWidth: 188, imgHeight: 275},
  {title:"Argylle", year: 2024, imgSrc: "/soundtracks/argylle.jpg", imgWidth: 188, imgHeight: 275},
  {title:"The Flash", year: 2023, imgSrc: "/soundtracks/flash.jpg", imgWidth: 188, imgHeight: 275},
  {title: "Downton Abbey", year: 2019, imgSrc: "/soundtracks/downton-abbey.jpg", imgWidth: 188, imgHeight: 275},
  {title: "Downton Abbey: A New Era", year: 2022, imgSrc: "/soundtracks/downton2.jpg", imgWidth: 188, imgHeight: 275},
  {title: "Enola Holmes", year: 2020, imgSrc: "/soundtracks/enola.jpg", imgWidth: 188, imgHeight: 275},
  {title: "Spider-Man: Across the Spider-verse", year: 2023, imgSrc: "/soundtracks/spider-man.jpg", imgWidth: 188, imgHeight: 275},
  {title: "Ferrari", year: 2023, imgSrc: "/soundtracks/ferrari.jpg", imgWidth: 188, imgHeight: 275},
  {title: "Shetland: series 8", year: 2023, imgSrc: "/soundtracks/shetland.jpg", imgWidth: 188, imgHeight: 275},
  {title: "Belgravia", year: 2020, imgSrc: "/soundtracks/belgravia.jpg", imgWidth: 188, imgHeight: 275},
  {title: "Project Artemis", year: 2024, imgSrc: "/soundtracks/artemis.jpg", imgWidth: 188, imgHeight: 275},
  {title: "The Reluctant Traveller (Series 2)", year: 2024, imgSrc: "/soundtracks/traveller.jpg", imgWidth: 188, imgHeight: 275},
  {title: "Gotham Knights", year: 2022, imgSrc: "/soundtracks/gotham.jpg", imgWidth: 188, imgHeight: 275},
  {title: "Devs", year: 2020, imgSrc: "/soundtracks/devs.jpg", imgWidth: 188, imgHeight: 275},
  {title: "Aldi Christmas Commercial", year: 2023, imgSrc: "/soundtracks/aldi.jpg", imgWidth: 270, imgHeight: 180},
  {title: "Tar", year: 2022, imgSrc: "/soundtracks/tar.jpg", imgWidth: 188, imgHeight: 275},
  {title: "Sigur Ros: √Åtta", year: 2023, imgSrc: "/soundtracks/atta.jpg", imgWidth: 275, imgHeight: 275},
]