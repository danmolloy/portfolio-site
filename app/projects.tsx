import { projectsArr, projectsBlurb } from "@/lib/projects";
import { FaGithub, FaLaptopCode, FaLink, FaNpm } from "react-icons/fa";
import Image from "next/image";
import Link from "next/link";
import Project from "./projects/[project]/project";


export default function Projects() {

  return (
    <section data-testid="projects-section" className="w-[95vw] py-12 m-2 flex flex-col items-center  min-h-[80vh]  ">
      <div className="flex flex-col w-full px-12 py-4 text-sm items-center">
        <h2 className=" ">Projects</h2>
      </div>
      <div className="flex flex-col sm:flex-row flex-wrap w-full justify-center items-center">
        {projectsArr.map((i, index) => (
            <Project key={index} project={i} />))}
      </div>
    </section>
  )
}