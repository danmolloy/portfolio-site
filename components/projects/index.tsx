import { useState } from "react";
import ProjectDiv from "./projectDiv";
import { Element } from "react-scroll";
import { ProjectDivProps, projectsBlurb } from "@/lib/projects";

export type ProjectsSectionProps = {
  stackArray: string[]
  projectsArr: ProjectDivProps[]
}

export default function ProjectsSection(props: ProjectsSectionProps) {
  const [selectedStack, setSelectedStack] = useState<string>("")
  const { stackArray, projectsArr } = props;

  const handleStackBtn = (lib: string) => {
    if (selectedStack === lib) {
      setSelectedStack("")
    } else {
      setSelectedStack(lib)
    }
  }

  return (
    <Element data-testid="projects-section" name="projects" className="w-[95vw] py-12 m-2 flex flex-col items-center  min-h-[80vh]  lg:w-2/3">
      <div className="flex flex-col w-full px-12 py-4 text-sm items-center">
      <h1 className="font-display ">Projects</h1>
      <p className="my-4 text-zinc-500">{projectsBlurb}</p>
      </div>
      <div data-testid="filters-arr" className="dark:bg-zinc-800 rounded flex flex-row flex-wrap items-center justify-center sm:w-2/3">
        {stackArray.map((i: string) => (
          <button onClick={() => handleStackBtn(i)} 
          className={selectedStack === i 
            ? "transition ease-out duration-300 text-white bg-blue-500 text-sm rounded m-1 " 
            : "transition ease-out duration-300 text-blue-500 text-sm rounded m-1 hover:text-blue-700 active:text-blue-800 "
          } key={i}>{i}</button>
        ))}
      </div>
      {selectedStack === ""
      ? projectsArr.map((i) => (
          <ProjectDiv key={i.id} {...i} />
        ))
      : projectsArr.filter(i => i.stack.includes(selectedStack)).map((i) => (
          <ProjectDiv key={i.id}  {...i} />
        ))}
    </Element>
  )
}