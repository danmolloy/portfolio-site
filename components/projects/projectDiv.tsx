import { useState } from "react";
import { ProjectDivProps } from "@/lib/projects";
import ProjectHeader from "./header";
import ProjectImages from "./images";

export default function ProjectDiv(props: ProjectDivProps) {
  const [selectedTab, setSelectedTab] = useState<number>(0)
  const { title, overview, features, stack, githubLink, previewLink, images } = props;

  return (
    <div data-testid="project-div" className="flex flex-col justify-center items-center my-8 border dark:border-zinc-700 p-4 rounded shadow w-full">
      <ProjectHeader {...{title, selectedTab, setSelectedTab, githubLink, previewLink}} />
      <div data-testid="project-text" className="transition duration-700">
      {selectedTab === 0 
      ? <div data-testid="overview-tab" className="w-full h-[50vh] md:px-8 md:h-[40vh] overflow-scroll">
        <div>
          {overview.map((i, index) => (
            <p className="pb-4" key={index}>{i}</p>
          ))}
          </div>
        </div>
      : <div data-testid="features-tab" className=" h-[50vh]  items-center md:px-8 md:h-[30vh]">
          {features.map(i => (
          <p className="" key={i}>{`- ${i}`}</p>
        ))}
        <div className="mt-4">
          {stack.map(i => (
            <p className="my-2" key={i}>
              {i}
            </p>
          ))}
        </div>
        </div>}
      </div>
      <ProjectImages images={images} />
      </div>
      )
}