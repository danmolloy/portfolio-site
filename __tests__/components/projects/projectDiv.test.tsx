import { act, fireEvent, render, screen } from "@testing-library/react";
import ProjectDiv from "@/components/projects/projectDiv";
import { ProjectDivProps, projectsArr } from "@/lib/projects";

const randInd = Math.floor(Math.random() * projectsArr.length)

const mockProps: ProjectDivProps = projectsArr[randInd]

describe("<ProjectDiv />", () => {
  beforeEach(() => {
    render(<ProjectDiv {...mockProps} />)
  })
  it("project-div is in the document", () => {
    const projectDiv = screen.getByTestId("project-div")
    expect(projectDiv).toBeInTheDocument()
  })
  it("project-header is in the document", () => {
    const projectHeader = screen.getByTestId("project-header")
    expect(projectHeader).toBeInTheDocument()
  })
  it("overview and stack are in the document", () => {
    const text = screen.getByTestId("project-text")
    expect(text).toBeInTheDocument()
    expect(text.textContent).toMatch(mockProps.overview[0])
    expect(text.textContent).toMatch(mockProps.stack[0])
  })
  it("features renders on features btn click, same with overview btn", () => {
    const featuresBtn = screen.getByText("Features")
    act(() => {
      fireEvent.click(featuresBtn)
    })
    const text = screen.getByTestId("project-text")
    expect(text.textContent).toMatch(mockProps.features[0])
    const overviewBtn = screen.getByText("Overview")
    act(() => {
      fireEvent.click(overviewBtn)
    })
    expect(text.textContent).toMatch(mockProps.overview[0])
  })
  it("project-images is in the document", () => {
    const projectImages = screen.getByTestId("project-images")
    expect(projectImages).toBeInTheDocument()
  })
  it("matches snapshot", () => {
    const projectDiv = screen.getByTestId("project-div")
    expect(projectDiv).toMatchSnapshot()
  })
})