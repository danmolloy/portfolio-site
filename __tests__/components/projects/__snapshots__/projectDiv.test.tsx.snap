// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectDiv /> matches snapshot 1`] = `
<div
  class="flex flex-col justify-center items-center my-8 outline"
  data-testid="project-div"
>
  <div
    data-testid="project-header"
  >
    <h2>
      MyJournal
    </h2>
    <a
      class="m-1 ml-2 hover:underline flex flex-row items-center"
      data-testid="github-link"
      href="https://github.com/danmolloy/django-journal"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M255.03 261.65c6.25 6.25 16.38 6.25 22.63 0l11.31-11.31c6.25-6.25 6.25-16.38 0-22.63L253.25 192l35.71-35.72c6.25-6.25 6.25-16.38 0-22.63l-11.31-11.31c-6.25-6.25-16.38-6.25-22.63 0l-58.34 58.34c-6.25 6.25-6.25 16.38 0 22.63l58.35 58.34zm96.01-11.3l11.31 11.31c6.25 6.25 16.38 6.25 22.63 0l58.34-58.34c6.25-6.25 6.25-16.38 0-22.63l-58.34-58.34c-6.25-6.25-16.38-6.25-22.63 0l-11.31 11.31c-6.25 6.25-6.25 16.38 0 22.63L386.75 192l-35.71 35.72c-6.25 6.25-6.25 16.38 0 22.63zM624 416H381.54c-.74 19.81-14.71 32-32.74 32H288c-18.69 0-33.02-17.47-32.77-32H16c-8.8 0-16 7.2-16 16v16c0 35.2 28.8 64 64 64h512c35.2 0 64-28.8 64-64v-16c0-8.8-7.2-16-16-16zM576 48c0-26.4-21.6-48-48-48H112C85.6 0 64 21.6 64 48v336h512V48zm-64 272H128V64h384v256z"
        />
      </svg>
      <p
        class="ml-1"
      >
        View GitHub
      </p>
    </a>
    <a
      class="m-1 hover:underline flex flex-row items-center"
      data-testid="example-link"
      href="https://django-journal-production.up.railway.app/"
      target="_blank"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
        />
      </svg>
      <p
        class="ml-1"
      >
        Live Example
      </p>
    </a>
    <div
      class="w-full flex flex-row mb-6"
    >
      <button
        class="custom-border-gradient border-b border-blue-500  py-2 my-2 mx-4 hover:text-zinc-400 "
        data-testid="overview-btn"
      >
        Overview
      </button>
      <button
        class="py-2 my-2 mx-4 hover:text-zinc-400"
      >
        Features
      </button>
    </div>
  </div>
  <div
    data-testid="project-text"
  >
    <div
      class=""
      data-testid="overview-tab"
    >
      <div>
        <p>
          My journal is a basic journalling web app, created as a personal learning tool when I was studying Django and Python. Very simply, users create an account and create journal entries which are only visible to themselves.
        </p>
        <p>
          MyJournal is built with Python and Django. It connects to a SQLite3 database and uses Django packages such as Django Auth.
        </p>
      </div>
      <div>
        <p
          class=""
        >
          Python
        </p>
        <p
          class=""
        >
          Django
        </p>
        <p
          class=""
        >
          Django-Templates
        </p>
        <p
          class=""
        >
          PostgreSQL
        </p>
      </div>
    </div>
  </div>
  <div
    class=" m-4 h-[796px]  flex flex-row items-center justify-center"
    data-testid="project-images"
  >
    <button
      class="opacity-30 transition-opacity duration-500 text-xl hover:text-amber-500"
      data-testid="left-chevron"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 320 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
        />
      </svg>
    </button>
    <img
      alt="Index View"
      class="mx-16 fade-in transition-opacity duration-200"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="199"
      src="/_next/image?url=%2Fimages%2Fmyjournal-index.png&w=750&q=75"
      srcset="/_next/image?url=%2Fimages%2Fmyjournal-index.png&w=384&q=75 1x, /_next/image?url=%2Fimages%2Fmyjournal-index.png&w=750&q=75 2x"
      style="color: transparent;"
      width="360"
    />
    <button
      class="opacity-100 transition-opacity duration-700 text-xl hover:text-amber-500"
      data-testid="right-chevron"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 320 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
        />
      </svg>
    </button>
  </div>
</div>
`;
